core/src/main/java/hudson/cli/ClientAuthenticationCache.java

```diff
@@ -104,15 +109,25 @@ public Authentication get() {
             return Jenkins.ANONYMOUS;
         }
     }
-
+    
     /**
      * Computes the key that identifies this Hudson among other Hudsons that the user has a credential for.
      */
     @VisibleForTesting
     String getPropertyKey() {
         String url = Jenkins.getActiveInstance().getRootUrl();
         if (url!=null)  return url;
-        return Secret.fromString("key").getEncryptedValue();
+        
+        LOGGER.log(Level.WARNING, "The instance is not configured using a rootUrl, the key that represents your instance will not be stable");
+        if(randomId == null){
+            synchronized (this){
+                if(randomId == null){
```

Comment:
Double-checked locking, but probably harmless due to use of `volatile`. Still needlessly complicated I think.